lib/iex/lib/iex/pry.ex

```diff
@@ -615,10 +615,102 @@ defmodule IEx.Pry do
 
   # IEx backend for Kernel.dbg/2.
   @doc false
+  def dbg(ast, options, env)
+
+  def dbg({:|>, _meta, _args} = ast, options, %Macro.Env{} = env) when is_list(options) do
+    [first_ast_chunk | asts_chunks] = ast |> Macro.unpipe() |> chunk_pipeline_asts_by_line(env)
+
+    initial_acc = [
+      quote do
+        env = unquote(Macro.escape(env))
+        options = unquote(options)
+
+        options =
+          if IO.ANSI.enabled?() do
+            Keyword.put_new(options, :syntax_colors, IO.ANSI.syntax_colors())
+          else
+            options
+          end
+
+        env = Map.update!(env, :line, &(unquote(line_from_asts(first_ast_chunk)) || &1))
```

Comment:
You will know if the line is nil or not at compilation time... so you can either emit `env = %{env | line: new_line}` or `env = env`.

Reply:
Something like:

```suggestion
        env = unquote(line_from_asts(first_ast_chunk, env_var))
```

Where `env_var = quote(do: env)` as you did for `options_var`.

Reply:
Updated 👍